packages/core/src/runtime/utils.ts

```diff
@@ -23,14 +24,18 @@ export function removeBase(url: string): string {
   return rawRemoveBase(url, siteData.base);
 }
 
-export function isEqualPath(a: string, b: string) {
-  return withBase(normalizeHref(a)) === withBase(normalizeHref(b));
+export function isEqualPath(a: string, b: string, cleanUrls: boolean) {
+  return withBase(normalizeHref(a, cleanUrls)) === withBase(normalizeHref(b, cleanUrls));
+}
+
+export function useNormalizeHrefInRuntime(a: string){
```

Comment:
I think it's better to implement the normalize ability by a normal util function instead of a react hook.The reasons are as follows:

1. We don't need to call the `usePageData` hook because we can get siteData from `virtual-site-data` directly.

2. The normalize process is a high frequency operation in all kinds of components.The react hook has a limitation, that is, the hook cannot be placed in a conditional statement to ensure that the hook execution order is always consistent, otherwise there will be some rendering bugs.We can avoid the problem as much as possible.

3. The name of utils can still be `normalizeHref` to make it clear and easier to understand. We can just import it from `@/runtime` instead of `@rspress/shared`.

Reply:
I understand the first two points and have already made the necessary adjustments.

Regarding the third point, I comprehend your perspective, but I have a different opinion on it. My concern is that having two identical symbols in the codebase could lead to confusion. For instance, when someone types `normalizeHref` in a runtime component code, there is a possibility that VSCode automatically imports the symbol from either `@/runtime` or `@rspress/shared`. This could result in unexpected behavior and make it difficult to locate any bugs.

Therefore, I suggest creating two separate symbols for each function to avoid these potential issues.

Reply:
I agree with you.And the `useNormalizeHrefInRuntime` could be renamed to `normalizeHrefInRuntime` because it's no longer a react hook.